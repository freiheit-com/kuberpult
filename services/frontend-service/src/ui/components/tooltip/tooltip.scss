@use '@material/tooltip/styles';

.mdc-tooltip.mdc-tooltip--rich {
    // !important to overwrite the default styling of the Tooltip.
    // The `mdc-tooltip--rich` is restricted to be placed on one of the 4 corners of a component
    // https://github.com/material-components/material-components-web/blob/066d9439b7f1bf11426153573560b35462a3b9d5/packages/mdc-tooltip/README.md?plain=1#L281-L286
    left: 10px !important;
    width: $release-card-tooltip-width !important;
    .mdc-tooltip__surface {
        box-shadow: none;
        background: transparent;
        // add padding to allow for the Caret (the arrow)
        padding: 8px 0px 0px 0px;
        border-radius: $border-radius-medium;

        .mdc-tooltip__content {
            background: var(--mdc-theme-background);
            // add margin to allow for the Caret (the arrow)
            margin: 12px 0px 0px 0px;
            border: 1px solid $release-card-tooltip-border-color;
            border-radius: $border-radius-medium;
        }

        .mdc-tooltip__caret-surface-top {
            border: 1px solid $release-card-tooltip-border-color;
            border-radius: 0px;
            left: $release-card-tooltip-caret-shift-left;
        }
    }
}
