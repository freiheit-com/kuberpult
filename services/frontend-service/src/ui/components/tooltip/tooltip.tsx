/*This file is part of kuberpult.

Kuberpult is free software: you can redistribute it and/or modify
it under the terms of the Expat(MIT) License as published by
the Free Software Foundation.

Kuberpult is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
MIT License for more details.

You should have received a copy of the MIT License
along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.

Copyright 2023 freiheit.com*/
import { Tooltip as TooltipReact } from 'react-tooltip';

export const Tooltip = (props: { children: JSX.Element; tooltipContent: JSX.Element; id: string }): JSX.Element => {
    const { children, tooltipContent, id } = props;
    const delayHide = 50; // for debugging the css, increase this to 1000000

    // The React tooltip really wants us to use a href, but also we don't want to do anything on click:
    const href = '#';
    return (
        <div className={'tooltip-container'}>
            <a href={href} id={'tooltip' + id} data-tooltip-place="bottom" data-tooltip-delay-hide={delayHide}>
                {children}
            </a>
            <TooltipReact className={'tooltip'} anchorSelect={'#tooltip' + id} border={'solid 2px lightgray'}>
                {tooltipContent}
            </TooltipReact>
        </div>
    );
};
