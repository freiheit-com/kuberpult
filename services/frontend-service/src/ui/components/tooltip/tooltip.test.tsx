/*This file is part of kuberpult.

Kuberpult is free software: you can redistribute it and/or modify
it under the terms of the Expat(MIT) License as published by
the Free Software Foundation.

Kuberpult is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
MIT License for more details.

You should have received a copy of the MIT License
along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.

Copyright 2023 freiheit.com*/
import React from 'react';
import { render } from '@testing-library/react';
import { Tooltip } from './tooltip';

describe('Tooltip', () => {
    const getNode = (): JSX.Element => (
        <Tooltip content={<div>I'm inside the tooltip</div>} children={<button>click me</button>} id={'test-tooltip'} />
    );
    const getWrapper = () => render(getNode());

    it(`Renders a tooltip`, () => {
        // when
        const { container } = getWrapper();
        // then
        expect(container.firstChild).toMatchSnapshot();
    });
});
