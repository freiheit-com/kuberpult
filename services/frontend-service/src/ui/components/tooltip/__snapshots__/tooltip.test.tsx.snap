// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Renders a tooltip 1`] = `
<div
  class="tooltip-container"
>
  <a
    data-tooltip-delay-hide="50"
    data-tooltip-place="bottom"
    href="#"
    id="tooltiptest-tooltip"
  >
    <button>
      click me
    </button>
  </a>
</div>
`;
